R/gpdd.R

```diff
@@ -1,22 +1,42 @@
- 
 
-#' Download Tables from the GPDD Data
-#' @param table name of the table to be downloaded.
-#' @return the requested data.frame (as a tibble)
-#' @export
-#' @examples 
-#' gpdd("location")
-gpdd <- function(table = c("data", "main", "timeperiod", "location")){
-  
-  table <- match.arg(table)
-  
-  url <- switch(table,
-                location = "https://knb.ecoinformatics.org/knb/d1/mn/v2/object/df35b.239.1",
-                main = "https://knb.ecoinformatics.org/knb/d1/mn/v2/object/df35b.234.1")
-    
+#' Download Data from GPDD.
+#'
+#' @author Zihao Li, \email{zihao0104@@berkeley.edu}
+#' @param dataset_name A string of the name of the dataset.
+#' @return The requested dataset in tibble format.
+#' @examples
+#' download_gpdd('data')
+#' download_gpdd('timeperiod')
 
-      
-  readr::read_csv(url)
-  
+
+download_gpdd <- function(dataset_name =  c("data",
+                                            "main",
+                                            "timeperiod",
+                                            "taxon", 
+                                            "datasource", 
+                                            "biotope", 
+                                            "location")) {
+  dataset_name <- match.arg(dataset_name)
```

Comment:
This is a good start.  Now that I have had a chance to think about this, I think we should change a little bit about how this function works though.  

To make this package more like a data package, we should save the downloaded data using `save()` to location given by `user_data_dir()` from the `rappdirs` package.  Then we should have an `.onLoad()` function that will read the data in to the user workspace.

Because the data names are quite generic, we should probably rename each with a prefix `gpdd_`, e.g. `gpdd_data`, `gpdd_location`, etc.

Reply:
@AlexLi0104 we can strategize about this last part at our next meeting, meanwhile I think this is good to merge.  nice work.